bindings/gumjs/gumquickvalue.c

```diff
@@ -537,6 +538,42 @@ _gum_quick_args_parse (GumQuickArgs * self,
 
         break;
       }
+      case 'M':
+      {
+        GumMatchPattern * pattern;
+
+        if (JS_IsString (arg))
+        {
+          const char * str;
+
+          str = JS_ToCString (ctx, arg);
+          if (str == NULL)
+            goto propagate_exception;
+
+          pattern = gum_match_pattern_new_from_string (str);
+          JS_FreeCString (ctx, str);
+
+          if (pattern == NULL)
+            goto invalid_pattern;
+
+        }
+        else if (JS_IsObject (arg))
+        {
+          pattern = JS_GetOpaque (arg, core->match_pattern_class);
+          if (pattern == NULL)
+            goto expected_pattern;
+        }
+        else
+        {
+          goto expected_pattern;
+        }
+
+        gum_match_pattern_ref (pattern);
```

Comment:
This will create an extra reference for the case where it's constructed from a string. So this `ref()` should only happen for the case when it's constructed from an existing `MatchPattern`.